src/build/build0.scm

```diff
@@ -6,17 +6,25 @@
   (for-each display args)
   (newline))
 
-(define (compile modf)
-  (displayln "... compile " modf)
-  (let ((proc (open-process
-               (list path: (getenv "GERBIL_GSC" "gsc")
-                     arguments: (list "-cc-options" "--param max-gcse-memory=300000000" modf)
-                     stdout-redirection: #f))))
-    (if (not (zero? (process-status proc)))
-      (error "Compilation error; gsc exit with nonzero status" modf))))
+(define (start-gsc modf)
+  (displayln "... compile " modf) (force-output)
+  (open-process
+   (list path: (getenv "GERBIL_GSC" "gsc")
+         arguments: (list "-cc-options" "--param max-gcse-memory=300000000" modf)
+         stdout-redirection: #f)))
 
-(let ((files (open-input-file file-list)))
-  (let lp ()
-    (let ((next (read-line files)))
-      (if (not (eof-object? next))
-        (begin (compile next) (lp))))))
+(define (end-gsc proc)
+  (if (not (zero? (process-status proc)))
+    (error "Compilation error; gsc exit with nonzero status" proc))
+  (close-port proc))
```

Comment:
I think we should probably `close-port` regardless of whether it is an error exit or not; so save process status, close port, examine process status.